### PR TITLE
Save embedly thumbnail URL's

### PR DESCRIPTION
**Pull request by mbertrand**
#### What are the relevant tickets?
Partially addresses #926 

#### What's this PR do?
- Adds a `LinkMeta` model with attributes `url` and `thumbnail`
- Adds a nullable ForeignKey from `Post` to `LinkMeta` (but this isn't really used for anything yet).
- Sets the `Post.link_meta` foreign key value for link posts in `channels.api.sync_post_model`
- Calls `LinkMeta.objects.get_or_create` every time the `embedly.api.views.embedly_view` is invoked (ie when creating or viewing a link post)
- Calls `LinkMeta.objects.get_or_create` on `PostSerializer.create` and adds `thumbnail` to the `channels.serializers.BasePostSerializer` class.
- Adds the thumbnail (`post_link_thumbnail`) to the `search.serializers.ESPostSerializer` class so that it is included in the index.

#### How should this be manually tested?
- Create some link posts and text posts.  There should not be any obvious changes on the front end.  
- Check the REST API response for link posts and text posts (`/api/v0/posts/<id>).  There should be a `thumbnail` attribute for both, that should always be null for text posts, but equal to whatever thumbnail was returned by embedly for link posts.  (They will also be null for old link posts created before this PR, until the post detail page is viewed or a new link post is made with the same URL).
- Do an elasticsearch query on your new link posts and make sure that `post_link_thumbnail` is populated with the correct values.

### CODE REVIEW

File: channels/models.py [hunk 1]
```diff
@@ -98,12 +99,21 @@ def __str__(self):
         return f"{self.name}"
 
 
+class LinkMeta(TimestampedModel):
+    """
+    The thumbnail embedly provides for a particular URL
+    """
+    url = URLField(unique=True, max_length=2048)
+    thumbnail = URLField(blank=True, null=True, max_length=2048)
```
Comment (noisecapella) [inline channels/models.py]: Can you add a `__str__` method?

File: channels/utils.py [hunk 2]
```diff
@@ -244,3 +245,25 @@ def get_reddit_slug(permalink):
         str: the reddit slug for a submission
     """
     return list(filter(None, permalink.split('/')))[-1]
+
+
+def get_or_create_link_meta(url):
```
Comment (noisecapella) [inline channels/utils.py]: Can you add a `@transaction.atomic` decorator here?

Comment (mbertrand) [reply channels/utils.py]: done

File: channels/models.py (hunk 1 above)
Comment (mbertrand) [reply channels/models.py]: done

**Comment by mbertrand** [comment]
@noisecapella addressed comments and rebased with master to fix a travis failure

**Pull request merged**